src/ai/backend/test/tester/config.py

```diff
@@ -37,36 +43,122 @@ class LoginCredential(BaseModel):
     )
 
 
-class Image(BaseModel):
+class ImageConfig(BaseConfigModel):
     name: Optional[str] = Field(
         default=None,
         description="The Docker image to use for the test context.",
         examples=["cr.backend.ai/multiarch/python:3.13-ubuntu24.04"],
     )
+    architecture: Optional[str] = Field(
+        default=None,
+        description="The architecture of the session.",
+        examples=["x86_64"],
+    )
```

Comment:
Strictly speaking, this part should have the config and the internal dataclass separated.
That part is a bit cumbersome, so I haven't been attaching the Config postfix, but this makes me think about it a bit.